http/server_test.ts

```diff
@@ -219,15 +219,20 @@ test(async function requestBodyStreamWithTransferEncoding() {
 });
 
 test(function requestCookieHandling() {
-  const req = new ServerRequest();
+  // cache handling
+  let req = new ServerRequest();
   req.headers = new Headers();
   assertEquals(req.cookie, {});
   req.headers = new Headers();
   req.headers.set("Cookie", "foo=bar");
-  assertEquals(req.cookie, { foo: "bar" });
+  assertEquals(req.cookie, {});
```

Comment:
This seems wrong?

Reply:
To read this yes it can feel wrong but it's just to show the cache handling.
At first the `ServerRequest` is created and the headers are set. After this the `req.cookie` is called, which will try to parse the headers to find cookies. It can't find anything so `{}` is stored in `_cookie`. Next headers is set to `foo=bar` and we try to recall `req.cookie` and it returns `{}` which is right because we have already called it and it's cached. The only way to get `foo=bar` is to recreate a `ServerRequest` like shown after in the test. Why do this? Because you don't set request cookie header when you try to access it, it's already set by the client request. It was just to cache the cookie parsing, especially in case of web server frameworks and middleware which can call this a lot while processing a request.